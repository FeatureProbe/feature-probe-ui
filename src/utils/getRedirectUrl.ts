import localForage from 'localforage';
import { getProjectList } from 'services/project';
import { IProject, IEnvironment } from 'interfaces/project';

export const getRedirectUrl = async (defaultRedirectUrl: string) => {
  let redirectUrl = defaultRedirectUrl;
  const projectKey: string = await localForage.getItem('projectKey') || '';
  const environmentKey: string = await localForage.getItem('environmentKey') || '';

  if (!!projectKey && !!environmentKey) {
    redirectUrl = `/${projectKey}/${environmentKey}/toggles`;
  } else {
    const res = await getProjectList<IProject[]>();
    const { success, data } = res;

    if (success && data) {
      const { key, environments } = data[0];
      let envKey = '';
      environments.some((environment: IEnvironment) => {
        if (environment.key) {
          envKey = environment.key;
          return true;
        }
        return environment;
      });

      if (key && envKey) {
        redirectUrl = `/${key}/${envKey}/toggles`;
      }
    }
  }

  return redirectUrl;
}