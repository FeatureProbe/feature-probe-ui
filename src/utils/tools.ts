import isString from 'lodash/isString';
import trim from 'lodash/trim';
import { IVariation } from 'interfaces/targeting';

export const replaceSpace = (obj: any) => {
  for (let k in obj) {
    if (obj[k] && isString(obj[k])) {
      obj[k] = trim(obj[k]);
    }
  }

  return obj;
}

export const isJSON = (str: string) => {
  try {
    if (JSON.parse(str) instanceof Object) {
      return true;
    }
  } catch (error) {
    return false;
  }
}

export const getVariationName = (variations: IVariation[]) => {
  let name = '';
  for (let i = 1; i <= 20; i++) {
    const index = variations.findIndex((variation) => {
      return variation.name === 'variation' + i;
    });
    if (index === -1) {
      name = 'variation' + i;
      break;
    }
  }
  return name;
}
