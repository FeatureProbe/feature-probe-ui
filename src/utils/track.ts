import ReactGA from 'react-ga4';

export const EventTrack = {
  init() {
    ReactGA.initialize('G-CMLDTZH6RC');
  },

  setUserId(userId: string) {
    ReactGA.set({ userId });
  },

  pageView(pathName: string) {
    ReactGA.send({ hitType: "pageview", page: pathName });
  },

  track(category: string, action: string) {
    ReactGA.event({
      category,
      action,
    });
  },
};
