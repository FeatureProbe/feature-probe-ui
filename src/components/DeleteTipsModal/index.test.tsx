import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import DeleteTipsModal from '.';
import { IntlWrapper } from 'components/utils/wrapper';

it('DeleteTipsModal snapshot', (done) => {
  (async () => {
    const { asFragment } = render(
      <DeleteTipsModal
        open={true}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
        content={'test content'}
        title={'test title'}
      />,
      {
        wrapper: IntlWrapper,
      }
    );
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

it('DeleteTipsModal clcik', (done) => {
  (async () => {
    const mockOnCancel = jest.fn();
    const mockOnConfirm = jest.fn();
    const { asFragment } = render(
      <DeleteTipsModal
        open={true}
        onCancel={mockOnCancel}
        onConfirm={mockOnConfirm}
        content={'test content'}
        title={'test title'}
      />,
      {
        wrapper: IntlWrapper,
      }
    );
    
    await userEvent.click(screen.getByText('Confirm'));
    await userEvent.click(screen.getByText('Cancel'));

    expect(mockOnCancel).toBeCalled();
    expect(mockOnConfirm).toBeCalled();

    done();
  })();
});
