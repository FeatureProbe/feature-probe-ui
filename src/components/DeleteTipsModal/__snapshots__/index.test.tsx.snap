// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTipsModal snapshot 1`] = `<DocumentFragment />`;
